array.c

```diff
@@ -4292,22 +4302,42 @@ ary_resize_smaller(VALUE ary, long len)
 
 /*
  *  call-seq:
- *     ary.delete(obj)            -> item or nil
- *     ary.delete(obj) {block}    -> item or result of block
+ *    array.delete(obj) -> deleted_object
+ *    array.delete {|obj| ... } -> deleted_object or block_return
```

Comment:
This should be `array.delete(obj) {|nosuch| … }` (or replace `nosuch` with another name as Array#delete requires an argument

Reply:
Fixed in next PR.